### DISCUSSION TITLE
Using IfcOpenShell modules for commercial software!

### DISCUSSION THREAD

**Discussion by PrimusEng** [IfcOpenShell/IfcOpenShell#2995] [General]
Hi IfcOpenShell,
we are developing software and would like top know if we can use freely your software libraries in our software?
Thanks for reply,
Primoz

**Comment by aothms** [comment]
Hi,

IfcOpenShell is licensed under LGPL, which is the most liberal of the "copy-left" licenses. This means that:

- when distributing your software to others (use behind a web-platform does not count as distributing):
  - your modifications (if any) to the IfcOpenShell library need to be made available
  - users need to be given the ability to freely update the version of IfcOpenShell used in your software, this typically means to use dynamic linking: .dll, .dylib, .so
